docs/overview/deep_dive/data_types.rst

```diff
@@ -467,33 +512,168 @@ Data Type Casting Modes
+dtype group, i.e, if the unsupported dtype encountered is :code:`uint8` , then this mode will try to upcast it to the next available supported :code:`uint` dtype. If no
+higher `uint` dtype is avaiable, then there won't be any upcasting performed. You can set this mode by calling :code:`ivy.upcast_data_types()` with an optional :code:`val` keyword argument that defaults to :code:`True`.
 
-Similarly, `downcast_data_dtypes` tries to downcast to the next lower supported dtype in the same dtype group. No casting is performed is no lower dtype is found in the same group.
-It can also be set by calling `ivy.downcast_data_types()` with the optional `val` keyword that defaults to boolean value `True`.
+Similarly, :code:`downcast_data_dtypes` tries to downcast to the next lower supported dtype in the same dtype group. No casting is performed is no lower dtype is found in the same group.
+It can also be set by calling :code:`ivy.downcast_data_types()` with the optional :code:`val` keyword that defaults to boolean value :code:`True`.
 
-`crosscast_data_types` is for cases when a function doesn't support `int` dtypes, but supports `float` and vice-versa. In such cases,
-we cast to the default supported `float` dtype if it's the unsupported integer case or we cast to the default supported `int` dtype if it's the unsupported `float` case.
+:code:`crosscast_data_types` is for cases when a function doesn't support :code:`int` dtypes, but supports :code:`float` and vice-versa. In such cases,
+we cast to the default supported :code:`float` dtype if it's the unsupported integer case or we cast to the default supported :code:`int` dtype if it's the unsupported :code:`float` case.
 
-The `cast_data_types` mode is the combination of all the three modes that we discussed till now. It works it way from crosscasting to upcasting and finally to downcasting to provide support
+The :code:`cast_data_types` mode is the combination of all the three modes that we discussed till now. It works it way from crosscasting to upcasting and finally to downcasting to provide support
 for any unsupported dtype that is encountered by the functions.
 
+This is the unsupported dtypes for :code: `exmp1`. It doesn't support :code: `float16`. We will see how we can
+still pass :code:`float16` arrays and watch it pass for different modes.
+
+Example of Upcasting mode :
+
+.. code-block:: python
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16", "complex")}, backend_version)
+    @handle_numpy_arrays_in_specific_backend
+    def expm1(x: torch.Tensor, /, *, out: Optional[torch.Tensor] = None) -> torch.Tensor:
+        x = _cast_for_unary_op(x)
+        return torch.expm1(x, out=out)
+
+The function :code:`expm1` has :code:`float16` as one of the unsupported dtypes, for the version :code:`2.0.1` which
+is being used for execution at the time of writing this. We will see how cating modes handles this.
+
+.. code-block:: python
+
+    import ivy
```

Comment:
I think we should avoid using `try-except` in example code-blocks, let's instead add a comment to the right of the function call, like, 
```
ret = ivy.expm1(ivy.array([1], dtype='float16'))    # raises Exception
ivy.upcast_data_types()
ret = ivy.expm1(ivy.array([1], dtype='float16'))    # doesn't raise an exception
```
Same applies to the other examples as well. We wouldn't need the code block for the output of each example either, as a result